modules/oke/locals.tf

```diff
@@ -6,10 +6,11 @@ locals {
   # used by cluster
   lb_subnet = var.preferred_load_balancer == "public" ? "pub_lb" : "int_lb"
 
-  ad_names = [
-    for ad_name in data.oci_identity_availability_domains.ad_list.availability_domains :
-    ad_name.name
-  ]
+  ad_number_to_name = {
+    for ad in data.oci_identity_availability_domains.ad_list.availability_domains :
+    ad.ad_number => ad.name
+  }
+  ad_numbers = keys(ad_number_to_name)
```

Comment:
This should be referenced as `local.ad_number_to_name)` once you've figured out how to resolve the above problem